code-challenges.test.js

```diff
@@ -14,33 +16,136 @@
+
+    // Return the properly grammarized string
+    return str1
+  }
+
+  // Capitalize first letter of each word
+  var y = array1.map(value => capitalizeAllFirstLetters(value.name))
+
+  // Concat capitalized name, + " is " + occupation
+  y = y.map( (value, index) => value + " is " + array1[index].occupation+".")
+
+  // Return an array with a sentence about each person
+  return y
+}
 
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
 // a) Create a test with an expect statement using the variables provided.
+// Errors encountered:
+// ReferenceError: remainders is not defined
+
+describe ('remainders', () => {
+  test('takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3', () => {
+
+  const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+  // Expected output: [ 2, 0, -1, 0 ]
+  const expectedOutput1 = [ 2, 0, -1, 0 ]
+
+  const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+  // Expected output: [ 2, 1, -1 ]
+  const expectedOutput2 = [ 2, 1, -1 ]
 
-const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
-// Expected output: [ 2, 0, -1, 0 ]
-const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
-// Expected output: [ 2, 1, -1 ]
+  expect(remainders(hodgepodge1)).toEqual(expectedOutput1)
+  expect(remainders(hodgepodge2)).toEqual(expectedOutput2)
+  })
+})
+
+// Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
 
 // b) Create the function that makes the test pass.
+// A function that takes in an array of mixed data types
+const remainders = (mixedArray) => {
+  // Filter array to only numbers
+  const filtered = mixedArray.filter(value => typeof value == 'number')
+  // Return each number modulus 3
+  return filtered.map(value => value % 3)
+}
+
 
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 // a) Create a test with an expect statement using the variables provided.
+// Errors received:
+// ReferenceError: cubeTheSum is not defined
+
+describe ('cubeTheSum', () => {
+  test('takes in an array of numbers and returns the sum of all the numbers cubed', () => {
 
-const cubeAndSum1 = [2, 3, 4]
-// Expected output: 99
-const cubeAndSum2 = [0, 5, 10]
-// Expected output: 1125
+    const cubeAndSum1 = [2, 3, 4]
+    // Expected output: 99
+    const expectedOutput1 = 99
+
+    const cubeAndSum2 = [0, 5, 10]
+    // Expected output: 1125
+    const expectedOutput2 = 1125
+
+  expect(cubeTheSum(cubeAndSum1)).toEqual(expectedOutput1)
+  expect(cubeTheSum(cubeAndSum2)).toEqual(expectedOutput2)
+  })
+})
+
+// Test Suites: 1 passed, 1 total
+// Tests:       3 passed, 3 total
 
 // b) Create the function that makes the test pass.
+// A function that takes in an array
+const cubeTheSum = (arrayNum) => {
```

Comment:
Your logic and tests look excellent in this section.  Your pseudocode does a decent job describing the steps, but there is definitely room to expand further.  For instance, quite a few times you are using a form of iteration and do not speak on that in your pseudocode.